### PR TITLE
Fix redis test

### PR DESCRIPTION
**Pull request by DeepthiYV**
- Related ticket: https://progress.opensuse.org/issues/111128
- Needles: NO
- Verification run: 
 SLE 15 SP3 - [x86_64](https://openqa.suse.de/tests/9274968#step/redis/22) | [s390x|](https://openqa.suse.de/tests/9274969#step/redis/22) | [aarch64](https://openqa.suse.de/tests/9274970#step/redis/22)
SLE 15 SP2 -  [x86_64](https://openqa.suse.de/tests/9274974#step/redis/22) | [s390x|](https://openqa.suse.de/tests/9274972#step/redis/22) | [aarch64](https://openqa.suse.de/tests/9274971#step/redis/22)
SLE 15 SP4 -  [x86_64](https://openqa.suse.de/tests/9274979#step/redis/22) | [ppc64le|](https://openqa.suse.de/tests/9274978#step/redis/22) | [aarch64](https://openqa.suse.de/tests/9274976#step/redis/22) | [svirt-xen-hvm](https://openqa.suse.de/tests/9274980#step/redis/17) | [svirt-xen-pv](https://openqa.suse.de/tests/9274981#step/redis/17) | [s390x](https://openqa.suse.de/tests/9274998#step/redis/22)
LEAP: [x86_64](http://10.161.229.147/tests/3801#step/redis/22)
Tumbleweed: [x86_64](http://10.161.229.147/tests/3800#step/redis/22)

### CODE REVIEW

File: data/console/redis_cli.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+else
+        echo 'Failed: redis-cli command 'redis-cli get foo' failed to execute'
+fi
+
+# Load a test db from the data directory
+out=$(redis-cli -h localhost -p 6379 < ./movies.redis)
+exec_redis 'redis-cli HMGET movie:343 title' 'Spider-Man'
+
+# Connect to the new instance of redis-server with port 6380 and  make 6380 instance 
+#a replica of redis instance running on port 6379 and verify the reslt
```
Comment (rfan1) [inline data/console/redis_cli.sh:47]: minor comments:

and  make 6380 -> delete the blank

verify the reslt -> result?

File: tests/console/redis.pm [hunk 2]
```diff
@@ -34,46 +27,25 @@ sub run {
...
-    wait_still_screen(5);
-
-    # test replication
-    validate_script_output('redis-cli info replication', sub { m/connected_slaves:1/ });
-    validate_script_output('redis-cli -p 6380 info replication', sub { m/role:slave/ });
-    validate_script_output('redis-cli -p 6380 HMGET "movie:343" title', sub { m/Spider-Man/ });
+sub run_script {
+    my $script = shift;
+    my $logfile = "output.txt";
+    record_info($script, "Running python script: $script");
```
Comment (rfan1) [inline tests/console/redis.pm:42]: IMO, it should be a shell script, but please omit my comment if I am wrong

File: data/console/redis_cli.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+sleep 10 
+
+out=$(redis-cli info replication)
+
+if [[ "$out" =~ "connected_slaves:1" ]]
+then
+        echo "redis-cli info replication on master"
+else
+        echo "Failed: redis-cli info replication on master"
+
```
Comment (rfan1) [inline data/console/redis_cli.sh:59]: delete this line

File: data/console/redis_cli.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+fi
+out=$(redis-cli -p 6380 info replication)
+
+if [[ "$out" =~ "role:slave" ]]
+then
+        echo "Role:slave in redis-cli on replica"
+else
+        echo "Failed: redis-cli info replication on replica"
+
```
Comment (rfan1) [inline data/console/redis_cli.sh:68]: delete this line

Comment (DeepthiYV) [reply data/console/redis_cli.sh]: Deleted. Thanks Richard

File: data/console/redis_cli.sh (hunk 1 above)
Comment (DeepthiYV) [reply data/console/redis_cli.sh]: Correction is done. Thanks for the  heads up

**Review by rfan1 — APPROVED** [review]

File: data/console/redis_cli.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+exec_redis 'redis-cli -p 6380 replicaof 127.0.0.1 6379' 'OK'
+sleep 10 
+
+out=$(redis-cli info replication)
+
+if [[ "$out" =~ "connected_slaves:1" ]]
+then
+        echo "redis-cli info replication on master"
+else
+        echo "Failed: redis-cli info replication on master"
```
Comment (foursixnine) [inline data/console/redis_cli.sh:61]:
```suggestion
        echo "Failed: redis-cli info replication on master, slave took longer than 10 seconds to show up"
```

File: data/console/redis_cli.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        exit 1 
+fi
+out=$(redis-cli -p 6380 HMGET "movie:343" title)
+if [ "$out" = "Spider-Man" ]
+then 
+	echo "redis-cli replicaof done" 
+else	
+	echo "Failed: redis-cli replication on replica "
+        exit 1
+fi
```
Comment (foursixnine) [inline data/console/redis_cli.sh:80]: Can you create a follow up ticket to rework this in the future?, i.e: add an extra check to know the status of the replication (basically, know if the replica is in sync with the master)

**Review by foursixnine — APPROVED** [review]

**Review by foursixnine — APPROVED** [review]
Aside from the niptick of one of the messages, everything looks good

File: data/console/redis_cli.sh (hunk 6 above)
Comment (DeepthiYV) [reply data/console/redis_cli.sh:80]: created ticket: https://progress.opensuse.org/issues/115592

**Pull request merged**